samples/subsys/fs/fat_fs/boards/sam4e_xpro.conf

```diff
@@ -0,0 +1,3 @@
+CONFIG_DISK_DRIVER_SDMMC=y
+CONFIG_SDHC=y
+CONFIG_SAM_HSMCI=y
```

Comment:
You should not need to explicitly enable the driver using this overlay, see my comment about how the `SAM_HSMCI` Kconfig should be defined